src/core/common/tlvs.cpp

```diff
@@ -128,6 +128,8 @@ otError Tlv::Find(const Message &aMessage, uint8_t aType, uint16_t *aOffset, uin
 
             VerifyOrExit(sizeof(ExtendedTlv) <= remainingLen);
             aMessage.Read(offset, sizeof(ExtendedTlv), &extTlv);
+
+            VerifyOrExit(extTlv.GetLength() <= (remainingLen - sizeof(ExtendedTlv)));
```

Comment:
Maybe I am missing something here?
We have `VerifyOrExit(size <= remainingLen);` right after this which I think would check for same.

Reply:
Ah! Figured it out! Issue with wrapping of `uint16_t` length...